kolibri/logger/api.py

```diff
@@ -71,24 +72,13 @@ def update(self, request, *args, **kwargs):
             raise Http404
         serializer = self.get_serializer(instance, data=request.data, partial=partial)
         serializer.is_valid(raise_exception=True)
-        self.perform_update(serializer)
-
-        if getattr(instance, '_prefetched_objects_cache', None):
-            # If 'prefetch_related' has been applied to a queryset, we need to
-            # forcibly invalidate the prefetch cache on the instance.
-            instance._prefetched_objects_cache = {}
-        default_response = dict(request.data)
-        # First look if the computed fields to be updated are listed:
-        updating_fields = getattr(serializer.root, 'update_fields', None)
-        # If not, fetch all the fields that are computed methods:
-        if updating_fields is None:
-            updating_fields = [field for field in serializer.fields if getattr(serializer.fields[field], 'method_name', None)]
-        for field in updating_fields:
-            method_name = getattr(serializer.fields[field], 'method_name', None)
-            if method_name:
-                method = getattr(serializer.root, method_name)
-                default_response[field] = method(instance)
-        return Response(default_response)
```

Comment:
We do, there are some loggers that require a response with the logged data, including some computed fields, after the serialization saves the field.

Reply:
I'm not sure that we actually need this in the frontend at any point though, as things get updated optimistically in the frontend (would need some double checking of this assumption though).

Reply:
If things have not changed since I did the serializers optimization, we needed it for some loggers. As an example, the MasteryLogSerializer response includes totalattempts and pastattempts that are calculated querying the Attemptlog table. If now these values are calculated in the frontend, we can discard that code. If not, it's still needed.

Reply:
Yeah, the totalattempts and pastattempts are optimistically updated on the frontend, so I think we can not worry about returning them.